src/RobustHomotopy/robust_homotopy_method.jl

```diff
@@ -100,10 +100,14 @@ function _second_order_newton_step(homHess::HomotopyHessian,
     solve!(hSolver, δ)
     δ .*= -1
 
-    # PERF: the line search is taking up 80%+ of the time in _second_order_newton_step
-    # evidently I need a better (or better optimized) line search.
-    # (The line search is non-allocating, except for the @warn message.)
-    α_star = line_search(x, time_step, homHess, t_k, δ)
+    # Create objective function
+    ϕ = α -> F_value(homHess, t_k, x + α * δ, time_step)
```

Comment:
The problem with this is that it creates the function at run time so the code can't really be compiled.